### PR TITLE
Reorganize the tests class

### PR DESCRIPTION
**Pull request by sundapeng**
1. Reorganize the tests class: 
 - moved all Unit tests of OpensslCipher to `OpensslCipherTest`
 - moved all Unit tests of stream to `AbstractCryptoStreamTest`
 - `CTRCryptoStreamTest` would test `CTRCryptoInputStream/CTRCryptoOutputStream`
 - `CryptoStream*Test` would test `CryptoInputStream/CryptoOutputStream`
1. Add slf4j simple for Unit tests, otherwise it will print a warning during maven test

### CODE REVIEW

File: src/test/java/com/intel/chimera/stream/AbstractCryptoStreamTest.java [hunk 1]
```diff
@@ -282,4 +294,171 @@ private int readAll(InputStream in, byte[] b, int offset, int len)
...
+    try {
+      return (Cipher)ReflectionUtils.newInstance(
+          ReflectionUtils.getClassByName(cipherClass), props, transformation);
+    } catch (ClassNotFoundException cnfe) {
+      throw new IOException("Illegal crypto cipher!");
+    }
+  }
+
+  @Test
+  public void testJceAesCtrCryptoCipher() throws Exception {
```
Comment (jerrychenhf) [inline src/test/java/com/intel/chimera/stream/AbstractCryptoStreamTest.java]: Shall we consider these tests are read write tests comparing the previous tests are read tests, write tests or skip tests?
If in this concept, we can name these test methods with proper name for easy understanding the purpose of these tests.

File: src/test/java/com/intel/chimera/stream/CryptoStreamCBCNoPaddingTest.java [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 import com.intel.chimera.cipher.CipherTransformation;
 
-public class CBC_NOPADDINGStreamTest extends AbstractCryptoStreamTest {
+public class CryptoStreamCBCNoPaddingTest extends AbstractCryptoStreamTest {
```
Comment (jerrychenhf) [inline src/test/java/com/intel/chimera/stream/CryptoStreamCBCNoPaddingTest.java]: Naming convention suggests to: CBCNoPaddingCryptoStreamTest which is identical to CTRCryptoStreamTest and AbstractCryptoStreamTest

File: src/test/java/com/intel/chimera/stream/CryptoStreamCTRNoPaddingTest.java [hunk 3]
```diff
@@ -21,7 +21,7 @@
 
 import com.intel.chimera.cipher.CipherTransformation;
 
-public class CTRStreamTest extends AbstractCryptoStreamTest {
+public class CryptoStreamCTRNoPaddingTest extends AbstractCryptoStreamTest {
```
Comment (jerrychenhf) [inline src/test/java/com/intel/chimera/stream/CryptoStreamCTRNoPaddingTest.java]: Suggests the name to CTRNoPaddingCryptoStreamTest

**Comment by jerrychenhf** [comment]
Thanks Dapeng for these big improvements on organizing the unit tests.

Tests for CBCPKCS5Padding is still missing.

File: src/test/java/com/intel/chimera/stream/AbstractCryptoStreamTest.java (hunk 1 above)
Comment (sundapeng) [inline src/test/java/com/intel/chimera/stream/AbstractCryptoStreamTest.java]: Good suggestion, I will change it to `testByteBufferByMixingCiphers()`

File: src/test/java/com/intel/chimera/stream/CryptoStreamCBCNoPaddingTest.java (hunk 2 above)
Comment (sundapeng) [inline src/test/java/com/intel/chimera/stream/CryptoStreamCBCNoPaddingTest.java]: Good suggestion, I will fix it on next update.

File: src/test/java/com/intel/chimera/stream/CryptoStreamCTRNoPaddingTest.java (hunk 3 above)
Comment (sundapeng) [inline src/test/java/com/intel/chimera/stream/CryptoStreamCTRNoPaddingTest.java]: Good suggestion, I will fix it on next update.

**Comment by sundapeng** [comment]
Thank @jerrychenhf for your review, I have updated the patch, I also added Unit Tests for PKCS5Padding and a fix for `int read()`

File: src/main/java/com/intel/chimera/stream/CryptoInputStream.java [hunk 4]
```diff
@@ -117,7 +116,15 @@ public CryptoInputStream(
 
   @Override
   public int read() throws IOException {
-    return (read(oneByteBuf, 0, 1) == -1) ? -1 : (oneByteBuf[0] & 0xff);
```
Comment (jerrychenhf) [inline src/main/java/com/intel/chimera/stream/CryptoInputStream.java]: The code might be more compact. and the while logic not depending on the initial value of "ret". For example:

int n;
while ( (n = read(oneByteBuf, 0, 1)) == 0);
return (n == -1) ? -1 : oneByteBuf[0] & 0xff;

Comment (sundapeng) [inline src/main/java/com/intel/chimera/stream/CryptoInputStream.java]: Good suggestion.

**Comment by sundapeng** [comment]
Thank @jerrychenhf for your comments, I have updated the patch. I also updated some methods' name in Unit Tests.

**Comment by jerrychenhf** [comment]
Looks good to ship. Thanks @sundapeng for the improving the tests.

**Comment by sundapeng** [comment]
Thank @jerrychenhf for your review!

**Pull request merged**